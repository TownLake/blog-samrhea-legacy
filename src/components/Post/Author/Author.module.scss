@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.author {
  border-top: 1px solid var(--color-page-border);
  max-width: $layout-post-width;
  @include padding-top(1);
  @include line-height(1);
  @include margin-top(1);
  @include margin-bottom(2);

  .bio {
    .twitter {
      display: block;
      text-decoration: underline;

      &:hover,
      &:focus {
        color: var(--color-typographic-link-s-font);
      }
    }
  }
}

@include breakpoint-sm {
  .author {
    margin-left: auto;
    margin-right: auto;
  }
}
