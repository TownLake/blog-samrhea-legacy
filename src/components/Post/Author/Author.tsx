import React from "react";

import { useSiteMetadata } from "@/hooks";
import { getContactHref } from "@/utils";

import * as styles from "./Author.module.scss";

const Author = () => {
  const { author } = useSiteMetadata();

  return (
    <div className={styles.author}>
      <p className={styles.bio}>
        {author.bio}
        <a
          className={styles.twitter}
          href={getContactHref("twitter", author.contacts.twitter)}
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>{author.name}</strong> on Twitter
        </a>
      </p>
    </div>
  );
};

export default Author;
