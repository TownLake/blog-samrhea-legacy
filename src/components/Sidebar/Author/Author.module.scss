@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.author {
  .photo {
    background-clip: padding-box;
    border-radius: 50%;
    display: inline-block;
    height: 75px;
    width: 75px;
    @include margin-bottom(0);

    img {
      border-radius: 50%;
    }
  }

  .title {
    font-size: $typographic-base-font-size * 1.125;
    font-weight: 600;
    margin: 0;
    @include line-height(1.125);

    .link {
      color: var(--color-typographic-base-font);

      &:hover,
      &:focus {
        color: var(--color-typographic-base-font);
      }
    }
  }

  .subtitle {
    color: $color-gray;
    @include line-height(1);
    @include margin-bottom(1);
  }

  .titleContainer {
    align-items: center;
    display: flex;
    justify-content: space-between;
    @include margin(0.5, 0, 0.5, 0);
  }
}
