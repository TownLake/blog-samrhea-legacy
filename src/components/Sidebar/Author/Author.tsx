import React from "react";

import { Link } from "gatsby";

import { Image } from "@/components/Image";
import { ThemeSwitcher } from "@/components/ThemeSwitcher";

import * as styles from "./Author.module.scss";

type Props = {
  author: {
    name: string;
    bio: string;
    photo: string;
  };
  isIndex?: boolean;
};

const Author = ({ author, isIndex }: Props) => (
  <div className={styles.author}>
    <Link to="/">
      <Image alt={author.name} path={author.photo} className={styles.photo} />
    </Link>

    <div className={styles.titleContainer}>
      {isIndex ? (
        <h1 className={styles.title}>
          <Link className={styles.link} to="/">
            {author.name}
          </Link>
        </h1>
      ) : (
        <h2 className={styles.title}>
          <Link className={styles.link} to="/">
            {author.name}
          </Link>
        </h2>
      )}
      <ThemeSwitcher />
    </div>
    <p className={styles.subtitle}>{author.bio}</p>
  </div>
);

export default Author;
