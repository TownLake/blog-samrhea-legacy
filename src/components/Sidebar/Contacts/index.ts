export { default as Contacts } from "./Contacts";
