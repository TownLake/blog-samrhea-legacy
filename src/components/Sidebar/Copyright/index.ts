export { default as Copyright } from "./Copyright";
