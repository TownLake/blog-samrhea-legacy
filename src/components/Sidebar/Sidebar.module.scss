@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;

  .inner {
    @include padding(1, 0.75, 0);
    position: relative;

    &:after {
      background: var(--color-sidebar-border);
      background: linear-gradient(
          to bottom,
          var(--color-sidebar-border) 0%,
          var(--color-sidebar-border) 48%,
          var(--color-sidebar-border-fade) 100%
      );
    }
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;

    .inner {
      @include padding(1.25, 0.75, 0);

      &:after {
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;

    .inner {
      @include padding-equal(1.5);
    }
  }
}
