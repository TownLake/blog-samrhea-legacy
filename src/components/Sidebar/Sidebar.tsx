import React from "react";

import { useSiteMetadata } from "@/hooks";

import { Author } from "./Author";
import { Contacts } from "./Contacts";
import { Copyright } from "./Copyright";
import { Menu } from "./Menu";

import * as styles from "./Sidebar.module.scss";

type Props = {
  isIndex?: boolean;
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();

  return (
    <div className={styles.sidebar}>
      <div className={styles.inner}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Copyright copyright={copyright} />
      </div>
    </div>
  );
};

export default Sidebar;
