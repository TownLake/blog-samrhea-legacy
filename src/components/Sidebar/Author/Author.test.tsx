import React from "react";

import { StaticQuery } from "gatsby";

import { Author } from "@/components/Sidebar/Author";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

const mockedStaticQuery = StaticQuery as jest.Mock;

describe("Author", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(() => null);
  });

  test("renders correctly", () => {
    const props = { isIndex: false, author: mocks.author };
    const tree = testUtils
      .createSnapshotsRenderer(<Author {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
