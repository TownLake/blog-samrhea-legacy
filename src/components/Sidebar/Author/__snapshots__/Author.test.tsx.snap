// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div>
  <a
    href="/"
  />
  <div>
    <h2>
      <a
        href="/"
      >
        John Doe
      </a>
    </h2>
  </div>
  <p>
    Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
  </p>
</div>
`;
